lib/spack/spack/cmd/mirror.py

```diff
@@ -65,6 +65,10 @@ def setup_parser(subparser):
         '-n', '--versions-per-spec',
         help="the number of versions to fetch for each spec, choose 'all' to"
              " retrieve all versions of each package")
+    create_parser.add_argument(
+        '--public', action='store_true',
+        help="if this is a public mirror, avoid adding packages when"
+             " licensing prohibits it")
```

Comment:
same here -- default to public, make private the special case so that it is not easy to accidentally violate a license.